lib/dataSource-url.js

```diff
@@ -211,6 +211,8 @@ kettle.dataSource.URL.handle.http = function (that, baseOptions, data) {
         method: "GET"
     };
     if (baseOptions.operation === "set") {
+        data = new Buffer(data);
```

Comment:
Can we have this assigned to a new variable, e.g. dataBuffer? Changing the type of an existing variable seems clean in languages like JavaScript but in practice harbours hazards for both humans and machines.